nativelib/src/main/scala/java/lang/ieee754tostring/ryu/RyuRoundingMode.scala

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018 Ulf Adams
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Credits:
+//
+//    1) This work is a heavily modified derivation of the original work by
+//       Ulf Adams at URL: https://github.com/ulfjack/ryu.
+//       As such, it inherits the Apache license of the original work.
+//       Thank you Ulf Adams.
+//
+//    2) The original java sources were converted to a rough draft of
+//       scala code using the service at URL: javatoscala.com.
+//
+//       The raw conversion did not compile and contained bugs due
+//       to the handling of break and return statements, but it saved
+//       days, if not weeks, of effort.
+//
+//       Thank you javatoscala.com.
+//
+//    3) All additional work, including introduced bugs,  is an original
+//       contribution to Scala Native development.
+
+package java.lang.ieee754tostring.ryu
+
+import scala.language.implicitConversions
+
+object RyuRoundingMode extends Enumeration {
+
+  val CONSERVATIVE: RyuRoundingMode = new conservativeRM()
+
+  val ROUND_EVEN: RyuRoundingMode = new roundEvenRM()
+
+  abstract class RyuRoundingMode {
+
+    def acceptUpperBound(even: Boolean): Boolean
+
+    def acceptLowerBound(even: Boolean): Boolean
+
+  }
```

Comment:
This looks like it's going to be better served by `final case class RyuRoundingMode(acceptLowerBound: Boolean, acceptUpperBound: Boolean)`.

Reply:
Sorry, I do not understand, especially for the ROUND_EVEN. Can you elaborate?

accept*Bound(even: Boolean) is a method, not a value. The methods in
rounding modes CONSERVATIVE and ROUND_EVEN differ, by intent & necessity.

Without violating SN, I am trying to stay pretty close to the Ryu original both because
I will probably have to re-port soon (order of months) and to allow a future maintainer
to compare the two Rosetta stone style.  Of course, have to follow SN closely enough
to get accepted. 

There is an argument that since only RoundingMode CONSERVATIVE is used in
the Ryu scala code, all the ROUND_EVEN stuff should be stripped out.  Since it
would be a straight deletion, any future maintainer should understand it. That
would allow the case class you recommended.,

I am still not sure what the implicit conversion does.  I think it is unnecessary.
In my sandbox, I am trying to see if I can get away deleting that conversion
and reducing the cognitive load all around.

Reply:
I was wrong, I somehow missed that those methods took parameters and assumed it was a configuration-like object which can be better expressed via case class here. Lets keep the code as is.